### DISCUSSION TITLE
How to add column styles into Table

### DISCUSSION THREAD

**Discussion by Jigna-Rathod** [exceljs/exceljs#1673] [Q&A]
## Description

Create a table using addtable method and add numFmt style property to columns, it does not not show any difference.
Also, if we try modify the table by getTable Method and then set the numFmt property in columns still no effect.

Lib version: X.Y.Z

## Steps To Reproduce

```javascript

let cols=[{name:'a',key:'a',style:{numFmt:'0.00'}},
               {name:'b',key:'b',style:{numFmt:'0.00'}},
               {name:'c',key:'c',style:{numFmt:'0.00'}}];

let rows=[[12.343,34.6565,56.773],
                [23.4564,54.565,12.567],
                ];
 worksheet.addTable({
      name: 'tableName',
      ref: 'E10',
      headerRow: true,
      style: {
        theme: TableStyleLight9,
        showRowStripes: false,
      },
      columns: cols,
       rows: rows,
    });
```

## The expected behaviour:

numFmt should be applied to columns

**Comment by Siemienik** [comment]
Hi @Jigna-Rathod 

**📬 Thank you for the mail message, I respond to you prioritically.**

Following the types definition file. There is no option to set style per table's column:

https://github.com/exceljs/exceljs/blob/2ab468b8bae2c8d5844f922d61ac7d960fee26be/index.d.ts#L1793-L1817

How ever, you can try to use `ws.getColumn('A').numFmt = `. It should works.


```js
ws.getColumn('A').numFmt='0.00';
ws.getColumn('B').numFmt='0.00';
ws.getColumn('C').numFmt='0.00';
```

**Reply by Siemienik** [reply]
@Jigna-Rathod Is that resolve your proplem?

**Reply by Jigna-Rathod** [reply]
@Siemienik  Thanks a lot for the suggestion. However, the requirement is to create dynamic tables at run time depending upon user selections and setting the numFmt. ws.getColumn('A'). does not have the column list that is created by addTable(). it has columns directly created in ws. Hence, waiting for this feature to be introduced. :)

Really appreciate your efforts !

**Comment by unimatrix27** [comment]
The documentation suggest that this is possible.  From the doc: 

`const table = ws.getTable('MyTable');

// Get Column Wrapper for second column
const column = table.getColumn(1);

// set some properties
column.name = 'Code';
column.filterButton = true;
column.style = {font:{bold: true, name: 'Comic Sans MS'}};`

But in fact this first generates a TS error because its not in the type definition, and if I TS-Ignore this, then it has no effect on the file

So I am wondering why is it in the documentation and what other workaround would be there to set styles for table columns

**Reply by maquejp** [reply]
The table.getColumn return also a TableColumn which is 

`export type TableColumn = Required<TableColumnProperties>`

Which doe not have a style property...

ps: I have just installed the package ^4.3.0

**Reply by Siemienik** [reply]
Oh, the typings for ExcelJS were never an advantage 😭  I have an idea to migrate the entire ExcelJS to TypeScript, however, I need to get financial influence for this target. 

For now, the best idea is to fix the lacking typings by creating PR and annotating me by @Siemienik to send me a notification.
